src/libraries/System.DirectoryServices.Protocols/tests/DirectoryServicesTestHelpers.cs

```diff
@@ -24,26 +27,43 @@ public static bool IsLibLdapInstalled
                 {
                     try
                     {
-                        // Attempt PInvoking into libldap
-                        IntPtr handle = ber_alloc(1);
-                        ber_free(handle, 1);
+                        // Attempt PInvoking into libldap on Linux
+                        IntPtr handle = ber_alloc_Linux(1);
+                        ber_free_Linux(handle, 1);
                         _isLibLdapInstalled = true;
                     }
                     catch (Exception)
                     {
-                        _isLibLdapInstalled = false;
+                        try
+                        {
+                            // Attempt PInvoking into libldap on OSX
+                            IntPtr handle = ber_alloc_OSX(1);
+                            ber_free_OSX(handle, 1);
```

Comment:
Why rely on first-chance-exception here on OSX rather than just cross-compile this file?

Reply:
We don't cross compile the tests any more. With this PR, we only have a net5.0 configuration for tests now which is why I rely on exceptions. Given the tests do target net5.0 though I could use the NativeLibrary trick just for the tests instead if you want me to.

Reply:
Yeah maybe that would make more sense and make this simpler. We do that for S.Drawing tests:

https://github.com/dotnet/runtime/blob/184f49b20ee9fc6c2e6649e22750cb6c54ffdc53/src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs#L46-L55

Reply:
I missed that this was for the tests.  I'm fine either way.  I like @safern's suggestion.